clustered/common/src/main/java/org/ehcache/clustered/common/PoolAllocation.java

```diff
@@ -72,6 +74,24 @@ public long getSize() {
     public String getResourceName() {
       return resourceName;
     }
+
+    @Override
+    public boolean isCompatible(PoolAllocation other) {
```

Comment:
The thing that I like about delegation is that it enforces the developer to add the compatibility check in every new `PoolAllocation` that gets added. If every check is in `ServerStoreCompatibility` things are distributed in multiple places and they someone might forget to keep the updates in sync(which is exactly what happened in our case). Of course through testing would expose that hole at some point. And personally I'm not a big fan of huge classes that handle a lot of logic in if-else blocks. But that's just personal opinion. If the rest of you are in favor of the "all in `ServerStoreCompatibility`" approach I'm not completely against that either.